src/withdrawal_limit/tests/withdrawal_limit_test.cairo

```diff
@@ -0,0 +1,222 @@
+use starknet::ContractAddress;
+use core::integer::BoundedInt;
+use snforge_std as snf;
+use snforge_std::{
+    ContractClassTrait, EventSpy, EventSpyTrait, EventsFilterTrait, EventSpyAssertionsTrait
+};
+use starknet_bridge::bridge::tests::constants::{
+    OWNER, L3_BRIDGE_ADDRESS, USDC_MOCK_ADDRESS, DELAY_TIME
+};
+use starknet_bridge::withdrawal_limit::{
+    mock::{
+        withdrawal_limit_mock, withdrawal_limit_mock::Event::WithdrawalEvent,
+        IMockWithdrawalLimitDispatcher, IMockWithdrawalLimitDispatcherTrait
+    },
+    interface::{IWithdrawalLimit, IWithdrawalLimitDispatcher, IWithdrawalLimitDispatcherTrait}
+};
+use starknet_bridge::withdrawal_limit::component::{
+    WithdrawalLimitComponent,
+    WithdrawalLimitComponent::{RemainingQuotaUpdated, DailyWithdrawalPercentageUpdated}
+};
+use openzeppelin::token::erc20::interface::{IERC20, IERC20Dispatcher, IERC20DispatcherTrait};
+
+fn deploy_withdrawal_limit() -> (IWithdrawalLimitDispatcher, EventSpy) {
+    let withdrawal_limit_mock_class_hash = snf::declare("withdrawal_limit_mock").unwrap();
+    let (withdrawal_limit_mock_address, _) = withdrawal_limit_mock_class_hash
+        .deploy(@array![])
+        .unwrap();
+    let withdrawal_limit_mock = IWithdrawalLimitDispatcher {
+        contract_address: withdrawal_limit_mock_address
+    };
+
+    let mut spy = snf::spy_events();
+    (withdrawal_limit_mock, spy)
+}
+
+
+pub fn deploy_erc20(name: ByteArray, symbol: ByteArray) -> ContractAddress {
```

Comment:
we already have this function in other test cases, can we not move to utils and reuse?

Reply:
Did not find a way to import files from the tests/ folder into src/ so had to replicate

Reply:
can you create an issue for this w/ an `enhancement` tag and add a todo on code as well